lib/common_test/src/test_server_ctrl.erl

```diff
@@ -3963,10 +3993,13 @@ run_test_case1(Ref, Num, Mod, Func, Args, RunInit,
     %% if the test case was executed sequentially, this updates the execution
     %% time count on the main process (adding execution time of parallel test
     %% case groups is done in run_test_cases_loop/4)
-    if is_number(Time) ->
-	    put(test_server_total_time, get(test_server_total_time)+Time);
-       true ->
-	    ok
+    case {Time, RetVal} of
+		_ when is_number(Time) ->
+			put(test_server_test_functions_execution_time, get(test_server_test_functions_execution_time)+Time);
```

Comment:
I think code would be easier to read if we add internal helper functions doing:
1. `put(test_server_test_functions_execution_time, Time)`
2.  `get(test_server_test_functions_execution_time)`

also `test_server_test_functions_execution_time` is a lot of reading/writing which is probably needed to be precise, but consider reducing `test_server` to `ts` which is kind of popular acronym around repo.